### ISSUE TITLE
sudo -i chdir fails sometimes with SELinux enforcing

### ISSUE THREAD

**Issue by neutronscott** [sudo-project/sudo#160]
```
[user@server ~]$ sudo -r unconfined_r -t unconfined_t -i -u oracle bash
sudo: unable to change directory to /home/oracle: Permission denied
sudo: unable to execute /bin/bash: Permission denied
[user@server ~]$ sudo -r unconfined_r -t unconfined_t -s -u oracle bash
[oracle@server user ]$ cd
[oracle@server ~ ]$ pwd
/home/oracle
[oracle@server ~ ]$
```
This is because /home/oracle is a different context than original user can access. SELinux role isn't set until later.

**Comment by millert** [comment]
The chdir needs to be deferred until sesh is run for the SELinux case.

**Comment by millert** [comment]
This should be fixed by [cf25035](https://github.com/sudo-project/sudo/commit/cf250354fca318ad5e44e98befb90cd83fc7cfe4)
